src/projection/node/create-projection-node.ts

```diff
@@ -748,16 +748,32 @@ export function createProjectionNode<I>({
             return boxWithoutScroll
         }
 
-        applyTransform(box: Box): Box {
+        applyTransform(box: Box, scrollOnly: boolean = false): Box {
             const withTransforms = createBox()
             copyBoxInto(withTransforms, box)
-            for (let i = 0; i < this.path.length; i++) {
+            for (
+                let i = scrollOnly ? this.path.length - 1 : 1;
+                i < this.path.length;
+                i++
+            ) {
                 const node = this.path[i]
+
+                if (
+                    node.options.shouldMeasureScroll &&
+                    node.scroll &&
+                    node !== node.root
+                ) {
+                    transformBox(withTransforms, {
+                        x: -node.scroll.x,
+                        y: -node.scroll.y,
+                    })
+                }
+
                 if (!hasTransform(node.latestValues)) continue
                 transformBox(withTransforms, node.latestValues)
             }
 
-            if (hasTransform(this.latestValues)) {
+            if (!scrollOnly && hasTransform(this.latestValues)) {
```

Comment:
I feel it makes the function a bit too complicated to keep track of what we should/should not do when scroll only, should we create a new helper for it?